src/test/java/com/sphy/hotelmanagementapplication/service/HotelServiceTest.java

```diff
@@ -0,0 +1,412 @@
+package com.sphy.hotelmanagementapplication.service;
+
+import com.sphy.hotelmanagementapplication.converter.*;
+import com.sphy.hotelmanagementapplication.domain.Admin;
+import com.sphy.hotelmanagementapplication.domain.Hotel;
+import com.sphy.hotelmanagementapplication.domain.Room;
+import com.sphy.hotelmanagementapplication.dto.HotelDTO;
+import com.sphy.hotelmanagementapplication.dto.RoomDTO;
+import com.sphy.hotelmanagementapplication.repositories.AdminRepository;
+import com.sphy.hotelmanagementapplication.repositories.ClientRepository;
+import com.sphy.hotelmanagementapplication.repositories.HotelRepository;
+import com.sphy.hotelmanagementapplication.repositories.RoomRepository;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class HotelServiceTest {
+
+
+        @Mock
+        RoomRepository roomRepository;
+
+        @Mock
+        HotelRepository hotelRepository;
+
+        @Mock
+        ClientRepository clientRepository;
+
+        @Mock
+        AdminRepository adminRepository;
+
+        @Mock
+        RoomService roomService;
+
+        @Mock
+        HotelService hotelService;
```

Comment:
Do not mock the service you test, check the RoomServiceTest - you need to create it with new keyword